modules/search/class.jetpack-search.php

```diff
@@ -282,6 +283,28 @@ public function load_assets() {
 		}
 	}
 
+	/**
+	 * Is search supported on the current plan
+	 *
+	 * @since 6.0
+	 * Loads scripts for Tracks analytics library
+	 */
+	public function is_search_supported() {
+		return Jetpack::active_plan_supports( 'search' );
```

Comment:
I think this is going to be a problem since this method is now deprecated; we switched to `Jetpack_Plan` some time ago, but that class does not exist on WordPress.com yet.

D25738-code never made it to wpcom, hence the differences.

Reply:
We override this in class.wpcom-search.php and only that class ever gets instantiated so it is not a problem.

The code runs ok on wp.com.

Reply:
Won't it be a problem in Jetpack though, since `Jetpack::active_plan_supports()` is deprecated here?
https://github.com/Automattic/jetpack/blob/2c5dba9c6ed6ba0076918f7b22491d143e9e1d3a/class.jetpack.php#L1561

Or am I misunderstanding what you're saying?

Reply:
I think I misunderstood. Why do you think this is a problem though? It looks like it calls Jetpack_Plan::supports() anyways. So better to keep using the deprecated function until the new stuff is supported on wp.com right?

Reply:
I brought the new class to wpcom in r200559-wpcom and updated `class.jetpack-search.php` to use it there. This way we can really be in sync.

Reply:
Could we opt to do this instead? This should be okay on  WordPress.com and in Jetpack:

```diff
diff --git a/modules/search/class.jetpack-search.php b/modules/search/class.jetpack-search.php
index a2e1e649a..e7466f8b4 100644
--- a/modules/search/class.jetpack-search.php
+++ b/modules/search/class.jetpack-search.php
@@ -154,7 +154,10 @@ class Jetpack_Search {
 	 * @since 5.0.0
 	 */
 	public function setup() {
-		if ( ! Jetpack::is_active() || ! $this->is_search_supported() ) {
+		if (
+			! Jetpack::is_active()
+			|| ( method_exists( 'Jetpack_Plan', 'supports' ) && ! Jetpack_Plan::supports( 'search' ) )
+		) {
 			/**
 			 * Fires when the Jetpack Search fails and would fallback to MySQL.
 			 *
@@ -283,16 +286,6 @@ class Jetpack_Search {
 		}
 	}
 
-	/**
-	 * Is search supported on the current plan
-	 *
-	 * @since 6.0
-	 * Loads scripts for Tracks analytics library
-	 */
-	public function is_search_supported() {
-		return Jetpack::active_plan_supports( 'search' );
-	}
-
 	/**
 	 * Does this site have a VIP index
 	 * Get the version number to use when loading the file. Allows us to bypass cache when developing.

```

Reply:
OK, I understand now, thanks for explaining. I think what I have just pushed should clean it all up.